### PR TITLE
Add unit tests for RCGetInteriorVehicleDataConsent

### PR DESCRIPTION
**Pull request by ValeriiMalkov**
#This PR is ready for review.

Risk

This PR makes no API changes.

Summary

This PR provides unit test coverage for RCGetInteriorVehicleDataConsent 

### CLA

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+using test::components::application_manager_test::MockCommandFactory;
+using test::components::policy_test::MockPolicyHandlerInterface;
+
+namespace rc_rpc_plugin_test {
+using namespace rc_rpc_plugin;
+
+namespace {
+const uint32_t kConnectionKey = 2u;
+const uint32_t kAppId = 5u;
+const int kModuleId = 153u;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc]: @ValeriiMalkov please use module id from static member of RCPlugin

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+        .WillByDefault(Return(true));
+    ON_CALL(mock_allocation_manager_, is_rc_enabled())
+        .WillByDefault(Return(true));
+  }
+
+  template <class Command>
+  application_manager::SharedPtr<Command> CreateRCCommand(
+      MessageSharedPtr& msg) {
+    InitCommand(kDefaultTimeout_);
+    return ::utils::MakeShared<Command>(msg ? msg : msg = CreateMessage(),
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc]: @ValeriiMalkov confusing implementation. So you can pass to CreateRCCommand null pointer, but any way some default command will be created. Please avoid such behavior

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+                                        app_mngr_,
+                                        rpc_service,
+                                        mock_hmi_capabilities_,
+                                        mock_policy_handler_,
+                                        mock_allocation_manager_);
+  }
+
+  MessageSharedPtr CreateBasicMessage() {
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc]: @ValeriiMalkov add comments. What avtualy basic message mean? What the dufferent between CreateBasicMessage and  CreateMessage ?

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    MessageSharedPtr message = CreateMessage();
+    (*message)[application_manager::strings::params]
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc]: @ValeriiMalkov you can extract additional reference variable params

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+  am::request_controller::RequestController request_controller;
+  am::rpc_service::RPCServiceImpl rpc_service;
+  utils::SharedPtr<RCAppExtension> rc_app_extention_;
+  utils::SharedPtr<am::plugin_manager::MockRPCPluginManager>
+      mock_rpc_plugin_manager;
+};
+
+TEST_F(RCGetInteriorVehicleDataConsentTest,
+       Run_MobileSendButtonPressMessage_HMISendASKDRIVERModeToMobile) {
+  MessageSharedPtr mobile_message = CreateBasicMessage();
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc]: @ValeriiMalkov please keep paradigm AAA (http://wiki.c2.com/?ArrangeActAssert) And, split this blocks with empty line

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+  ON_CALL(mock_policy_handler_, CheckModule(_, _)).WillByDefault(Return(true));
+  EXPECT_CALL(app_mngr_, GetPluginManager())
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc]: @ValeriiMalkov look like candidate for ON_CALL

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+      .WillRepeatedly(ReturnRef(*mock_rpc_plugin_manager));
+  utils::Optional<RPCPlugin> rpc_plugin(mock_rpc_plugin);
+  utils::Optional<MockRPCPlugin> mrpc_plugin(mock_rpc_plugin);
+  EXPECT_CALL(*mock_rpc_plugin_manager, FindPluginToProcess(_, _))
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc]: @ValeriiMalkov please move to ON_CALLS staff that is not related to logic that you are testing

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+      .WillOnce(Return(rc_rpc_plugin::AcquireResult::ASK_DRIVER));
+  EXPECT_CALL(*mrpc_plugin, GetCommandFactory())
+      .WillOnce(ReturnRef(mock_command_factory));
+  EXPECT_CALL(mock_command_factory, CreateCommand(_, _))
+      .WillOnce(Return(rc_consent_request));
+  EXPECT_CALL(mock_hmi_interfaces_,
+              GetInterfaceFromFunction(
+                  hmi_apis::FunctionID::RC_GetInteriorVehicleDataConsent))
+      .WillOnce(Return(am::HmiInterfaces::HMI_INTERFACE_RC));
+  application_manager::SharedPtr<commands::ButtonPressRequest> command =
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc]: @ValeriiMalkov use auto

Comment (ValeriiMalkov) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc]: @LuxoftAKutsan fixed at the commit 051264163471fb3bb21b58ace83bbb7f9cf89b84

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc (hunk 7 above)
Comment (ValeriiMalkov) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc]: @LuxoftAKutsan fixed at the commit 051264163471fb3bb21b58ace83bbb7f9cf89b84

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc (hunk 6 above)
Comment (ValeriiMalkov) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc]: @LuxoftAKutsan fixed at the commit 051264163471fb3bb21b58ace83bbb7f9cf89b84

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc (hunk 5 above)
Comment (ValeriiMalkov) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc]: @LuxoftAKutsan fixed at the commit 051264163471fb3bb21b58ace83bbb7f9cf89b84

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc (hunk 1 above)
Comment (ValeriiMalkov) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc]: @LuxoftAKutsan fixed at the commit 051264163471fb3bb21b58ace83bbb7f9cf89b84
